usmqe_tests/api/tendrl/others/test_user.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf8 -*-
+"""
+REST API test suite - user
+"""
+
+from usmqe.api.tendrlapi import user as tendrlapi_user
+
+
+"""@pylatest default
+Setup
+=====
+
+Prepare USM cluster accordingly to documentation.
+
+Further mentioned ``APIURL`` points to: ``http://USMSERVER/api/1.0``.
+"""
+
+"""@pylatest default
+Teardown
+========
+"""
+
+
+def test_user_get(valid_access_credentials):
+    """@pylatest api/user.get
+    API-users: get user
+    *******************
+
+    .. test_metadata:: author mkudlej@redhat.com dahorak@redhat.com fbalak@redhat.com
+
+    Description
+    ===========
+
+    Get user admin.
+    """
+    test = tendrlapi_user.ApiUser()
+    """@pylatest api/user.get
+    .. test_step:: 2
+
+        Send **GET** request to ``APIURL/users``.
+
+    .. test_result:: 2
+
+        List of users in database is returned
+    """
+    test.users(valid_access_credentials)
+    """@pylatest api/user.get
+    .. test_step:: 3
+
+        Get user info.
+
+        Send **GET** request to ``APIURL/users/admin``.
+
+    .. test_result:: 3
+
+        User information for user *admin* is returned.
+    """
+    test.user('admin', valid_access_credentials)
+
+
+def test_user_get_not_found(invalid_access_credentials, not_found_response):
+    """@pylatest api/user.get_nonexistent
+    API-users: get nonexistent user
+    *******************************
+
+    .. test_metadata:: author dahorak@redhat.com fbalak@redhat.com
+
+    Description
+    ===========
+
+    Get users.
+    """
+    test = tendrlapi_user.ApiUser()
+
+    """@pylatest api/user.get_nonexistent
+    .. test_step:: 2
+
+        Try to get information for non existent user.
+
+        For each user, send GET request to ``APIURL/user/USER``
+
+    .. test_result:: 2
+
+        It should return error about unknown user.
+    """
+    test.user(
+        invalid_access_credentials,
+        asserts_in=not_found_response)
+
+
+def test_user_add_del(valid_access_credentials, valid_user_import, not_found_response):
+    """@pylatest api/user.add_delete
+    API-users: add and delete
+    *************************
+
+    .. test_metadata:: author mkudlej@redhat.com dahorak@redhat.com fbalak@redhat.com
+
+    Description
+    ===========
+
+    Add and remove *test* user.
```

Comment:
I think we could figure this out later, it looks like a good start at this point.